pages/api/dashboard/index.ts

```diff
@@ -118,24 +119,29 @@ const getAllMonths = function (transactions: any[]): AllMonths {
 }
 
 const getUserDashboardData = async function (userId: string): Promise<DashboardData> {
-  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId)).length
+  const buttonsCount = (await paybuttonService.fetchPaybuttonArrayByUserId(userId))
   const addresses = await addressService.fetchAllUserAddresses(userId, true)
   const XECAddresses = addresses.filter((addr) => addr.networkId === XEC_NETWORK_ID)
   const BCHAddresses = addresses.filter((addr) => addr.networkId === BCH_NETWORK_ID)
   const BCHTransactions = Array.prototype.concat.apply([], BCHAddresses.map((addr) => addr.transactions))
   const XECTransactions = Array.prototype.concat.apply([], XECAddresses.map((addr) => addr.transactions))
   const incomingTransactionsInUSD = BCHTransactions.map((t) => {
     t.amount = t.amount.times(DUMMY_BCH_PRICE)
+    t.network = 'BCH'
+    t.button = buttonsCount.find(buttons => buttons.addresses.find(add => add.address.id === t.addressId))
```

Comment:
This is confusing, `buttons` here refer to a single button

Reply:
For sure, changed to button